### PR TITLE
Activate user account manually front

### PR DESCRIPTION
**Pull request by cabch**
- Migration task to add the user setting with account activation by default set to false
- New tab user in Technical settings menu with checkbox to enable/disable manual activation of user accounts

- Clicking the verification email link works ✅
- Email verification set user to inactive if user's `manual activation` setting is set to true / active otherwise works ✅
- Sending email to inform user that its account is validated / need admin validation works ✅

### CODE REVIEW

**Comment by LucasBrazi06** [comment]
@cabch Not linked to an issue

File: src/app/services/component.service.ts [hunk 1]
```diff
@@ -405,6 +405,41 @@ export class ComponentService {
...
+    // build settings to proceed update
+    const settingsToSave = {
+      id: settings.id,
+      identifier: settings.identifier,
+      content: {
+        type: UserSettingsContentType.USER,
+        user: {
+          autoActivateAccountAfterValidation: settings.autoActivateAccountAfterValidation,
+        }
+      }
```
Comment (ClaudeROSSI) [inline src/app/services/component.service.ts:441]: This is a bit risky. You may forget to save some settings if we have something ese to save than autoActivetae ... in the future.
To be on the safe side, please try to apply the same logic than the one used when  saving cryptoSettings (line #402):
==> content: Utils.cloneObject(settings),

Comment (ClaudeROSSI) [reply src/app/services/component.service.ts:441]: let's create an issue and postpone it.

**Review by ClaudeROSSI — APPROVED** [review]

File: src/app/pages/settings-technical/users/settings-user.component.html [hunk 2]
```diff
@@ -0,0 +1,26 @@
+<div>
+    <form class="form" [formGroup]="formGroup" height="100%">
```
Comment (LucasBrazi06) [inline src/app/pages/settings-technical/users/settings-user.component.html]: Tabs -> 2 spaces

File: src/app/authentication/verify-email/authentication-verify-email.component.ts [hunk 3]
```diff
@@ -128,8 +130,13 @@ export class AuthenticationVerifyEmailComponent implements OnInit, OnDestroy {
           // Go to reset password
           this.router.navigate(['auth/define-password'], { queryParams: { hash: this.resetToken } });
         } else {
-          // Show message
-          this.messageService.showSuccessMessage(this.messages['verify_email_success']);
+          if (response.userStatus && response.userStatus === UserStatus.INACTIVE) {
```
Comment (LucasBrazi06) [inline src/app/authentication/verify-email/authentication-verify-email.component.ts]: `if (response.userStatus && response.userStatus === UserStatus.INACTIVE) {`
\=
`if (response?.userStatus === UserStatus.INACTIVE) {`

File: src/app/types/Setting.ts [hunk 4]
```diff
@@ -3,14 +3,15 @@ import TenantComponents from './TenantComponents';
...
+  identifier: TenantComponents | TechnicalSettingsType;
   sensitiveData: string[];
   category?: 'business' | 'technical';
   content: SettingContent;
+  autoActivateAccountAfterValidation?: boolean;
 }
 
 export interface SettingContent {
-  type: CryptoSettingsType | RoamingSettingsType | AnalyticsSettingsType | RefundSettingsType | PricingSettingsType | BillingSettingsType | SmartChargingSettingsType | AssetSettingsType;
+  type: CryptoSettingsType | RoamingSettingsType | AnalyticsSettingsType | RefundSettingsType | PricingSettingsType | BillingSettingsType | SmartChargingSettingsType | AssetSettingsType | TechnicalSettingsType;
```
Comment (LucasBrazi06) [inline src/app/types/Setting.ts]: Changed TechnicalSettingsType -> UserSettingsType

File: src/app/types/Setting.ts [hunk 5]
```diff
@@ -240,3 +242,21 @@ export interface AssetGreencomConnectionType {
   clientId: string;
   clientSecret: string;
 }
+
+export enum TechnicalSettingsType {
+  USER = 'user',
+  //TODO: add CRYPTO = 'crypto' here
```
Comment (LucasBrazi06) [inline src/app/types/Setting.ts]: Renamed TechnicalSettingsType -> TechnicalSettings
I moved Crypto in this enum

File: src/app/types/Setting.ts [hunk 6]
```diff
@@ -240,3 +242,21 @@ export interface AssetGreencomConnectionType {
...
+}
+
+export enum UserSettingsContentType {
```
Comment (LucasBrazi06) [inline src/app/types/Setting.ts]: Renamed UserSettingsContentType -> UserSettingsType

File: src/app/types/Setting.ts [hunk 7]
```diff
@@ -240,3 +242,21 @@ export interface AssetGreencomConnectionType {
...
+  USER = 'user',
+}
+
+export interface UserSetting extends Setting {
+  identifier: TechnicalSettingsType.USER;
+  type: UserSettingsContentType;
+  user?: UserSettings;
+}
+export interface UserSettings {
```
Comment (LucasBrazi06) [inline src/app/types/Setting.ts:265]: Added blank line :-)

**Review by LucasBrazi06 — APPROVED** [review]
Just few comments to read, no action on your side

**Pull request merged**